src/state/machines/turn.ts

```diff
@@ -479,11 +480,45 @@ export const TurnMachine = setup({
         };
       }),
     ),
+    stageAbilityUseBlocked: assign(({ context }: { context: GameState }) =>
+      produce(context, (draft) => {
+        draft.stage = {
+          eventType: "abilityUseBlocked",
+          cause: context.blockers.ability.reasons,
+          effect: undefined,
+          outcome: "negative",
+        };
+      }),
+    ),
+    stageSkipTurn: assign(({ context }: { context: GameState }) =>
+      produce(context, (draft) => {
+        draft.stage = {
+          eventType: "skipTurn",
+          cause: context.blockers.turn.reasons,
+          effect: undefined,
+          outcome: "negative",
+        };
+      }),
+    ),
     unstage: assign(({ context }: { context: GameState }) =>
       produce(context, (draft) => {
         draft.stage = undefined;
       }),
     ),
+    blockAbilityUse: assign(({ context }: { context: GameState }, reasons: GamePiece["uid"]) =>
```

Comment:
What are these actions used for?

Reply:
Well... Had an idea to have kinda API thing. So we can invoke those `block/unblock` actions from exp. pack cards. Turned out it's quite hard to do. So they aren't used. I'll remove them

Reply:
Yeah I had the same issue using external actions, one thing we can try in the future is to have a TurnMachineActionImplementations const or something, that just contains the function body logic that we can share between the main machine and the expansion "machines", but not sure if that will solve much